packages/apputils/src/instancetracker.ts

```diff
@@ -334,36 +347,25 @@ export class InstanceTracker<T extends Widget>
 
     this._restore = options;
 
-    return Promise.all(promises)
-      .then(([saved]) => {
-        return Promise.all(
-          saved.ids.map((id, index) => {
-            const value = saved.values[index];
-            const args = value && (value as any).data;
-            if (args === undefined) {
-              return state.remove(id);
-            }
-
-            // Execute the command and if it fails, delete the state restore data.
-            return registry
-              .execute(command, args)
-              .catch(() => state.remove(id));
-          })
-        );
+    const [saved] = await Promise.all(promises);
+    const val = await Promise.all(
+      saved.ids.map((id, index) => {
+        const value = saved.values[index];
+        const args = value && (value as any).data;
+        if (args === undefined) {
+          return state.remove(id);
+        }
+
+        // Execute the command and if it fails, delete the state restore data.
+        return registry.execute(command, args).catch(() => state.remove(id));
       })
-      .then(val => {
-        this._restored.resolve(void 0);
-        return val;
-      });
+    );
+    this._restored.resolve(void 0);
```

Comment:
Since we're using `undefined` now, could we change this to `undefined` instead of `void 0`.

Also since `Promise.all` resolves with an array, could we call it `values` instead of a singular?